data-raw/Global_IRI_Historical_ASAP_active.R

```diff
@@ -0,0 +1,324 @@
+#'   Next Steps: once w decide on additional strata for analysis we can combine the polygon grid and the
+#'               new strata polygon.
+#' 
+#' 
+#' Note - to run in background:navigate to project root directory and run:
+#'   caffeinate -i -s Rscript data-raw/df_global_iri_active_thresholds.R
+
+
+
+# We initially ran this analysis using phenology data converted to binary using "End of Season" (EOS) as the definition
+# for ending the active season. We later decided it would be good to use "Senescence" (SEN) to create this rasters
+# Therefor, I can make this script flexible to take either raster input by just allowing the analyst to choose which set of rasters to use as input
+# all this does is change a single file path.
+
+phenology_data_definition <- c(
+  "EOS",
+  "SEN"
+)[2]
+
+
+# let user define what outputs to write, or rather just store in memory for testing
+write_grid <- c(T,F)[2] # already wrote this out
+
+write_merged_tabular_data <- c(T,F)[1]
+
+
+
+library(tidyverse)
+library(sf)
+library(terra)
+library(exactextractr)
+library(arrow)
+
+# Load file/gdb paths -----------------------------------------------------
+
+# define output locations/file paths
+fp_output_grid <-  file.path(
+  Sys.getenv("AA_DATA_DIR"),
+  "private",
+  "processed",
+  "glb",
+  "iri",
+  "iri_global_grid.shp"
+)
+fp_output_merged_tab <- file.path(
+  Sys.getenv("AA_DATA_DIR"),
+  "private",
+  "processed",
+  "glb",
+  "iri",
+  paste0("Global_IRI_Historical_ASAP_active_",
+         phenology_data_definition,".parquet")
+)
+
+# adm0 gdb
+gdb_adm0 <- file.path(
+  Sys.getenv("AA_DATA_DIR"),
+  "public",
+  "raw",
+  "glb",
+  "asap",
+  "reference_data",
+  "gaul0_asap_v04"
+)
+
+dir_iri <-  file.path(
+  Sys.getenv("AA_DATA_DIR"),
+  "private",
+  "processed",
+  "glb",
+  "iri",
+  "tif"
+)
+
+dir_phen_base <- file.path(
+  Sys.getenv("AA_DATA_DIR"),
+  "public",
+  "processed",
+  "glb",
+  "asap",
+  "season")
+
+if(phenology_data_definition=="EOS"){
+  dir_phen <-  file.path(
+    dir_phen_base,
+    "trimester_any"
+  ) 
+}
+
+if(phenology_data_definition=="SEN"){
+  dir_phen <-  file.path(
+    dir_phen_base,
+    "trimester_any_sen"
+  ) 
+}
+
+
+# Load data into memory ---------------------------------------------------
+
+# Global admin 0
+# `USE:`IRI data has overlapping pixels so is not spatially valid - therefore we will
+# clip i t to the bbox of all continents to remove this issue
+gdf_adm0 <- st_read(
+  gdb_adm0,
+  layer = "gaul0_asap"
+)
+
+# ASAP Phenology data - rasters containing binary values indicating whether or not
+r_phen_m3 <- rast(
+  list.files(dir_phen,full.names = T),
+)
+
+# a nice name lookup table for renaming raster bands
+# it's more complicated than necessary, but gives us a couple options for flexibility
+df_phen_lookup <- tibble(
+  file_name = names(r_phen_m3),
+  mo_combo = str_extract_all(file_name, "\\d.*") %>% 
+    unlist()
+) %>% 
+  separate(
+    mo_combo,into = c("mo1","mo2","mo3"),sep = "-"
```

Comment:
>This doesn't work for me because mo3 contains the substring _sen.

oh good catch i must have introduced this issue when I added the "EOS" "SEN prefixes later.

Reply:
fixed [0715dbf](https://github.com/OCHA-DAP/ds-global-monitoring-explore/pull/4/commits/0715dbfca88aa9ab18fa91aea4c652628f34e8cd)